### PR TITLE
Fix Oracle OCI builder ignores region value provided in OCI config file

### PR DESCRIPTION
**Pull request by prydie**
Only default the OCI builder region to `us-phoenix-1` when no value is present in the packer template _and_ the OCI config file.

Closes: #5401 

/cc @mwhooker @owainlewis

### CODE REVIEW

**Comment by prydie** [comment]
We are generally removing the defaulting of the OCI region from projects as it causes more confusion than it provides utility. I haven't removed the defaulting completely in this PR, however, as it now forms part of the released builder API and probably warrants some discussion.

/cc @rickard-von-essen @owainlewis

File: builder/oracle/oci/client/config_test.go [hunk 1]
```diff
@@ -81,7 +81,7 @@ func TestNewConfigDefaultsPopulated(t *testing.T) {
 		t.Fatal("Expected ADMIN config to exist in map")
 	}
 
-	if adminConfig.Region != "us-phoenix-1" {
+	if adminConfig.Region != "us-ashburn-1" {
 		t.Errorf("Expected 'us-phoenix-1', got '%s'", adminConfig.Region)
```
Comment (owainlewis) [inline builder/oracle/oci/client/config_test.go]: Expected 'us-ashburn-1'

File: builder/oracle/oci/config.go [hunk 2]
```diff
@@ -101,7 +101,10 @@ func NewConfig(raws ...interface{}) (*Config, error) {
 
 	if c.Region != "" {
 		accessCfg.Region = c.Region
-	} else {
+	}
+
+	// Default if the template nor the API config contains a region.
+	if accessCfg.Region == "" {
 		accessCfg.Region = "us-phoenix-1"
```
Comment (owainlewis) [inline builder/oracle/oci/config.go]: As an aside, we could move these into constants to prevent using strings everywhere.

Comment (prydie) [reply builder/oracle/oci/config.go]: I think we should do a follow up PR to remove the region defaulting (to match the change to the OCI terraform provider).

**Pull request merged**